### PR TITLE
[XLA:CPU][oneDNN] Align with oneDNN for supported data types in matmul

### PR DESCRIPTION
**Pull request by mdfaijul**
This PR updates the data type support based on CPU features and aligns with oneDNN.

### CODE REVIEW

File: xla/service/cpu/onednn_util.h [hunk 1]
```diff
@@ -17,25 +17,35 @@ limitations under the License.
...
+#include "xla/xla_data.pb.h"
 
 namespace xla {
 namespace cpu {
 
 inline bool IsSupportedType(xla::PrimitiveType dtype) {
   using tsl::port::CPUFeature;
-  static bool is_bf16_supported = TestCPUFeature(CPUFeature::AVX512_BF16) ||
-                                  TestCPUFeature(CPUFeature::AMX_BF16);
+  bool is_supported = false;
```
Comment (penpornk) [inline xla/service/cpu/onednn_util.h:28]: Nit: We can just get rid of this variable and just return the value/expression directly, e.g., `return true`, `return TestCPUFeature...`.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

File: xla/service/cpu/onednn_util.h (hunk 1 above)
Comment (mdfaijul) [reply xla/service/cpu/onednn_util.h]: Thanks! Corrected.

**Review by penpornk — APPROVED** [review]
Thank you for the changes!

**Pull request closed without merging**